object.go

```diff
@@ -578,55 +596,34 @@ func (obs *obs) Delete(name string) error {
 		return ErrBadObjectMeta
 	}
 
-	// Place a rollup delete marker.
+	// Place a rollup delete marker and publish the meta
 	info.Deleted = true
 	info.Size, info.Chunks, info.Digest = 0, 0, _EMPTY_
-
-	metaSubj := fmt.Sprintf(objMetaPreTmpl, obs.name, sanitizeName(name))
-	mm := NewMsg(metaSubj)
-	mm.Data, err = json.Marshal(info)
-	if err != nil {
-		return err
-	}
-	mm.Header.Set(MsgRollup, MsgRollupSubject)
-	_, err = obs.js.PublishMsg(mm)
-	if err != nil {
+	if err = publishMeta(obs, name, info); err != nil {
 		return err
 	}
 
-	// Purge chunks for the object.
-	chunkSubj := fmt.Sprintf(objChunksPreTmpl, obs.name, info.NUID)
-	return obs.js.purgeStream(obs.stream, &streamPurgeRequest{Subject: chunkSubj})
+	// Purge chunks for the object
+	return purgeChunks(obs, info.NUID)
 }
 
-// AddLink will add a link to another object into this object store.
+// AddLink will add a link to another object.
 func (obs *obs) AddLink(name string, obj *ObjectInfo) (*ObjectInfo, error) {
-	if obj == nil {
+	if obj == nil || obj.Name == "" || obj.Bucket == "" {
 		return nil, errors.New("nats: object required")
 	}
 	if obj.Deleted {
 		return nil, errors.New("nats: object is deleted")
 	}
-	name = sanitizeName(name)
-	if !keyValid(name) {
-		return nil, ErrInvalidObjectName
-	}
 
-	// Same object store.
-	if obj.Bucket == obs.name {
-		info := *obj
-		info.Name = name
-		if err := obs.UpdateMeta(obj.Name, &info.ObjectMeta); err != nil {
-			return nil, err
-		}
-		return obs.GetInfo(name)
-	}
-
-	link := &ObjectLink{Bucket: obj.Bucket, Name: obj.Name}
+	// create the meta object
```

Comment:
I don't know why the original code was trying to do something with same vs different bucket, but I'm pretty sure there is no need to.  I'm not even sure of the intention of the original 616 - 623 block the way it's coded, what was the intent here?

Reply:
Also, if we don't care if they put an object or link over an existing object, do we care if they link to a deleted object?

Reply:
Will need to spend some time here but right now have other priorities.